Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -52,48 +53,273 @@ public static class CustomEntitySearch
             }
 
             WebApiSkillResponse response = WebApiSkillHelpers.ProcessRequestRecords(skillName, requestRecords,
-                (inRecord, outRecord) => {
-                    string text = inRecord.Data["text"] as string;
-                    IList<string> words = ((JArray)inRecord.Data["words"]).ToObject<List<string>>();
-                    if (words == null)
+                 (inRecord, outRecord) => {
+                     if (!inRecord.Data.ContainsKey("text"))
+                     {
+                         outRecord.Errors.Add(new WebApiErrorWarningContract { Message = "The given key 'text' was not present in the dictionary." });
+                     }
+                     else if (!inRecord.Data.ContainsKey("words") && 
+                     (inRecord.Data.ContainsKey("synonyms") || inRecord.Data.ContainsKey("exactMatches") || inRecord.Data.ContainsKey("fuzzyMatchOffset")))
+                     {
+                         outRecord.Errors.Add(new WebApiErrorWarningContract { Message = "Cannot process record without the given key 'words' in the dictionary"});
```

Comment:
The error message doesn't match the actual test.

Reply:
nit: to slightly reduce the level of nesting, you can directly `return outRecord` from both those error cases, and do away with the final `else`.